### DISCUSSION TITLE
Sqlalchemy: psycopg2 paramstyle for session.execute()

### DISCUSSION THREAD

**Discussion by paulefoa** [sqlalchemy/sqlalchemy#8745] [Usage Questions] [answered]
So before 1.4 this was possible:
```
session.execute("select %(param1)s, %(param2)s; from example", params={'param1': "test", 'param': "test2"})
```

How would I do that now?
The same query gives this error now
```
ProgrammingError: (psycopg2.errors.SyntaxError) syntax error at or near "%"
LINE 1: select %(param1)s, %(param2)s from example;
               ^

[SQL: select %%(param1)s, %%(param2)s from example;]
```

I know that I can use `:param1` but I specifically need psycopg2 bindings, because my examply is a little bit more sophisticated and I need to generate the name of the table dynamically.

I've read this part https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#all-in-expressions-render-parameters-for-each-value-in-the-list-on-the-fly-e-g-expanding-parameters
Specifically about `render_postcompile` and `literal_binds` but they're applied to sqlalchemy query and I have just a string with sql.
Is there still a way to do this?

**Comment by CaselIT** [comment, marked as answer]
Hi,

> `session.execute("select %(param1)s, %(param2)s; from example", params={'param1': "test", 'param': "test2"})`

this will raise a warning in 1.4: https://github.com/sqlalchemy/sqlalchemy/blob/996063e5a87f64b4536d441eaa12ba2f8b765a44/lib/sqlalchemy/engine/base.py#L1357-L1363

So following the warning you would use `session.connection().exec_driver_sql()`

**Reply by paulefoa** [reply]
Thank you, this is exactly what I was looking for!
